LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan/Mic.cs

```diff
@@ -78,14 +78,14 @@ private static Mic ComputeForJoinRequest(byte[] keyBytes, MacHeader mhdr, JoinEu
             return new Mic(BinaryPrimitives.ReadUInt32LittleEndian(cmac));
         }
 
-        public static Mic ComputeForJoinAccept(AppKey appKey, MacHeader macHeader, Memory<byte> joinNonce, Memory<byte> netId, DevAddr devAddr, Memory<byte> dlSettings, Memory<byte> rxDelay, Memory<byte> cfList)
+        public static Mic ComputeForJoinAccept(AppKey appKey, MacHeader macHeader, AppNonce joinNonce, Memory<byte> netId, DevAddr devAddr, Memory<byte> dlSettings, Memory<byte> rxDelay, Memory<byte> cfList)
         {
-            var algoInput = new byte[MacHeader.Size + joinNonce.Length + NetId.Size + DevAddr.Size + dlSettings.Length + rxDelay.Length + cfList.Length];
+            var algoInput = new byte[MacHeader.Size + AppNonce.Size + NetId.Size + DevAddr.Size + dlSettings.Length + rxDelay.Length + cfList.Length];
             var index = 0;
             _ = macHeader.Write(algoInput.AsSpan(index));
             index += MacHeader.Size;
-            joinNonce.CopyTo(algoInput.AsMemory(index));
-            index += joinNonce.Length;
+            _ = joinNonce.Write(algoInput.AsSpan(index));
```

Comment:
Don't we have the order of the bytes reversed compared to what we had previously here (since it's a random value it probably won't matter)?

Reply:
Not really. Since we have a strong type that reads it correctly (in little endian) from and to buffers. The reversals [were](https://github.com/Azure/iotedge-lorawan-starterkit/pull/1227/files#diff-2e7394e990e11eedfe539de7db67c6c0ca0423cc09dea7714dfb864c6febbc80L92) [removed](https://github.com/Azure/iotedge-lorawan-starterkit/pull/1227/files#diff-2e7394e990e11eedfe539de7db67c6c0ca0423cc09dea7714dfb864c6febbc80L142). While it's a random value, the `MessageProcessorJoinTest.When_Device_Is_Found_In_Api_Should_Update_Twin_And_Return` test does fail on line if the byte order is incorrect.